source/core/parameters.cc

```diff
@@ -219,11 +240,95 @@ namespace Parameters
+
+      // Multiphasic simulations parameters definition
+      number_fluids = prm.get_integer("number of fluids");
+      for (unsigned int i_fluid = 0; i_fluid < number_fluids; ++i_fluid)
+        {
+          fluids[i_fluid].parse_parameters(prm, i_fluid);
+        }
+      // Compatibility from multiphasic to monophasic parameter definition
+      if (number_fluids == 1)
+        {
+          viscosity            = fluids[0].viscosity;
+          density              = fluids[0].density;
+          specific_heat        = fluids[0].specific_heat;
+          thermal_conductivity = fluids[0].thermal_conductivity;
+          tracer_diffusivity   = fluids[0].tracer_diffusivity;
+        }
+    }
+    prm.leave_subsection();
+  }
+
+  void
+  Fluid::declare_parameters(ParameterHandler &prm, unsigned int id)
+  {
+    prm.enter_subsection("fluid " + Utilities::int_to_string(id, 1));
+    {
+      prm.declare_entry("density",
+                        "1",
+                        Patterns::Double(),
+                        "Density for the fluid corresponding to Phase = " +
+                          Utilities::int_to_string(id, 1));
+      prm.declare_entry(
+        "kinematic viscosity",
+        "1",
+        Patterns::Double(),
+        "Kinematic viscosity for the fluid corresponding to Phase = " +
+          Utilities::int_to_string(id, 1));
+      prm.declare_entry(
+        "dynamic viscosity",
+        "0",
+        Patterns::Double(),
+        "Dynamic viscosity for the fluid corresponding to Phase = " +
+          Utilities::int_to_string(id, 1));
+      prm.declare_entry(
+        "specific heat",
+        "1",
+        Patterns::Double(),
+        "Specific heat for the fluid corresponding to Phase = " +
+          Utilities::int_to_string(id, 1));
+      prm.declare_entry(
+        "thermal conductivity",
+        "1",
+        Patterns::Double(),
+        "Thermal conductivity for the fluid corresponding to Phase = " +
+          Utilities::int_to_string(id, 1));
+      prm.declare_entry(
+        "tracer diffusivity",
+        "0",
+        Patterns::Double(),
+        "Tracer diffusivity for the fluid corresponding to Phase = " +
+          Utilities::int_to_string(id, 1));
+    }
+    prm.leave_subsection();
+  }
+
+  void
+  Fluid::parse_parameters(ParameterHandler &prm, unsigned int id)
+  {
+    prm.enter_subsection("fluid " + Utilities::int_to_string(id, 1));
+    {
+      density           = prm.get_double("density");
+      dynamic_viscosity = prm.get_double("dynamic viscosity");
+      viscosity         = prm.get_double("kinematic viscosity");
+      if (dynamic_viscosity == 0)
+        // if the dynamic_viscosity is not given in prm
+        dynamic_viscosity = density * viscosity;
+      if (dynamic_viscosity != 0 && viscosity == 1)
+        // if the dynamic_viscosity is given in prm, and not the kinematic
+        // viscosity
+        viscosity = dynamic_viscosity / density;
```

Comment:
I think it would be important to check that density > 0 :)!

Reply:
yes sure! Done ;)